server/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -987,33 +987,23 @@ public XContentBuilder rawField(String name, InputStream value, XContentType con
      */
     @Deprecated
     public XContentBuilder rawField(String name, BytesReference value) throws IOException {
-        generator.writeRawField(name, value);
+        generator.writeRawField(name, value.streamInput());
         return this;
     }
 
     /**
      * Writes a raw field with the given bytes as the value
      */
     public XContentBuilder rawField(String name, BytesReference value, XContentType contentType) throws IOException {
```

Comment:
Do you plan to zap this one in a followup?

Reply:
Yeah, this PR is just for the generator, the builder I'll address after this

I know it's lots of PRs, but it's much easier to do small changes to make sure nothing breaks while changing things